lib/librato/collector/counter_cache.rb

```diff
@@ -49,16 +53,13 @@ def flush_to(queue, opts={})
         @lock.synchronize do
           # work off of a duplicate data set so we block for
           # as little time as possible
-          counts = @cache.dup
+          # requires a deep copy of data set
+          counts = JSON.parse(@cache.dup.to_json, symbolize_names: true)
```

Comment:
> May should move the JSON parsing part of this outside of the lock. The goal here was to make this as fast as possible since while it is happening it blocks writes any counter writes completely.

JSON parsing occurs here since we need a deep dup instead of a dup. Should it still occur outside of the lock?